### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by kq333**
[DEMO LINK](https://kq333.github.io/layout_catalog/)
[TEST REPORT LINK](https://kq333.github.io/layout_catalog/report/html_report/)

### CODE REVIEW

**Comment by PufAnimalsDev** [comment]
if you delete one product card, the grid is not saved, Volodymyr talked about it in class

File: src/styles/header.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    display: flex;
+    align-items: center;
+    height: 60px;
+    font-family: "Roboto", sans-serif;
+    font-size: 12px;
+    font-weight: 500;
+    white-space: nowrap;
+    text-decoration: none;
+    text-transform: uppercase;
+    color: var(--color-txt-primary);
```
Comment (PufAnimalsDev) [inline src/styles/header.scss:56]: try using scss variables not css variables

File: src/styles/header.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  &__link.is-active {
+    color: var(--color-link);
+  }
+
+  &__link {
+    position: relative;
+    display: flex;
+    align-items: center;
+    height: 60px;
+    font-family: "Roboto", sans-serif;
```
Comment (PufAnimalsDev) [inline src/styles/header.scss:50]: repeating, you use this on an html tag

File: src/styles/product-card.scss [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  height: 56px;
+  display: flex;
+}
+
+.product-card__button-action-link {
+  background: var(--color-link);
+  border: transparent;
+  border-radius: 5px;
+  width: 166px;
+  height: 40px;
```
Comment (PufAnimalsDev) [inline src/styles/product-card.scss:117]: use padding insted of height

File: src/styles/stars.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    width: 16px;
+    height: 16px;
+    margin-right: 4px;
+    background-repeat: no-repeat;
+    background-position: center center;
+    background-image: url("../images/star.svg");
+  }
+}
+
+.stars--1 :nth-child(-n + 1),
```
Comment (PufAnimalsDev) [inline src/styles/stars.scss:15]: checklist number 12

File: src/styles/colors.scss [hunk 5]
```diff
@@ -0,0 +1,7 @@
+:root {
+  --color-bg: #fff;
+  --color-link: #00acdc;
```
Comment (PufAnimalsDev) [inline src/styles/colors.scss:3]:
```suggestion
  $color-link: #00acdc;
```

File: src/styles/product-card.scss [hunk 6]
```diff
@@ -0,0 +1,134 @@
+html {
```
[the remark this answers is not in the document]
Comment (PufAnimalsDev) [reply src/styles/product-card.scss:1]: repeating, you have these styles in the file: main-page.scss

File: src/styles/header.scss [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+  &__link.is-active {
+    color: var(--color-link);
+  }
+
+  &__link {
+    position: relative;
+    display: flex;
+    align-items: center;
+    height: 60px;
```
[the remark this answers is not in the document]
Comment (PufAnimalsDev) [reply src/styles/header.scss:49]: it could be a mistake, if you leave the height, that might be good too

**Review by PufAnimalsDev — APPROVED** [review]
one comment, I don't see any changes in stars.scss where the for loop should be, it's in checklist, it may not pass, I would suggest a change

**Comment by kq333** [comment]
> one comment, I don't see any changes in stars.scss where the for loop should be, it's in checklist, it may not pass, I would suggest a change

done.

File: src/styles/colors.css [hunk 8]
```diff
@@ -0,0 +1,9 @@
+:root {
```
[the remark this answers is not in the document]
Comment (d-wasilewski) [reply src/styles/colors.css:1]: That's true, but I see you have the scss file with variables already. You can delete this one then

File: src/styles/colors.scss
Comment (d-wasilewski) [inline src/styles/colors.scss:1]: again, this file might be removed, you have it it _colors if im not mistaken

File: src/styles/main.css
Comment (d-wasilewski) [inline src/styles/main.css:1]: remove this file

File: src/styles/main.css.map
Comment (d-wasilewski) [inline src/styles/main.css.map:1]: you can remove it too

File: src/styles/product-card.scss [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    width: 200px;
+    border: 1px solid $color-border-secondary;
+    border-radius: 5px;
+    background-color: $color-bg;
+  }
+
+  &__product-image {
+    width: 160px;
+    height: 134px;
+    padding-top: 32px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:16]: margin

File: src/styles/product-card.scss [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  &__description {
+    font-style: normal;
+    font-weight: 500;
+    font-size: 12px;
+    line-height: 18px;
+    color: $color-txt-primary;
+    width: 166px;
+    height: 36px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:28]: this shouldnt be fixed

File: src/styles/product-card.scss [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+    margin: 40px 16px 4px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:29]: use either top or bottom margins to be consistent

File: src/styles/product-card.scss (hunk 10 above)
Comment (d-wasilewski) [reply src/styles/product-card.scss:28]: Here and in other places

File: src/styles/product-card.scss [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+    border: 1px solid transparent;
+    border-radius: 5px;
+    color: $color-bg;
+    text-transform: uppercase;
+    font-weight: 700;
+    font-size: 14px;
+    line-height: 16px;
+    text-decoration: none;
+    padding: 10px 0;
+    margin: 16px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:93]: you shouldnt give a margin to each element here, you can just set padding on the container to create this space

File: src/styles/product-card.scss [hunk 13]
```diff
@@ -0,0 +1,105 @@
+.product-card {
+  width: 200px;
+  margin-right: 48px;
+  padding-bottom: 39px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:4]: It's different on figma, and it should be margin not padding

**Review by d-wasilewski — CHANGES_REQUESTED** [review]

File: src/styles/product-card.scss (hunk 11 above)
Comment (d-wasilewski) [reply src/styles/product-card.scss:30]: Please fix

File: src/styles/product-card.scss [hunk 14]
```diff
@@ -0,0 +1,107 @@
+.product-card {
+  width: 200px;
+  margin-right: 48px;
+  padding-bottom: 49px;
```
Comment (d-wasilewski) [inline src/styles/product-card.scss:4]: This should be margin. In general, you should use padding when you want to add space within an element and use margin when you want to add space outside an element.

File: src/styles/header.scss [hunk 15]
```diff
@@ -0,0 +1,74 @@
+.header {
+  padding: 0 50px;
```
[the remark this answers is not in the document]
Comment (michalcywka) [reply src/styles/header.scss:2]: +1

**Review by michalcywka — CHANGES_REQUESTED** [review]
See comments

**Review by michalcywka — CHANGES_REQUESTED** [review]
Please have a look at all the comments, not all of the issues are fixed

**Review by michalcywka — APPROVED** [review]

**Pull request closed without merging**